Source/WebKit/WebProcess/WebProcess.cpp

```diff
@@ -2471,6 +2476,22 @@ Ref<WebCookieJar> WebProcess::protectedCookieJar()
     return m_cookieJar;
 }
 
+#if ENABLE(CONTENT_EXTENSIONS)
+void WebProcess::setResourceMonitorContentRuleList(WebCompiledContentRuleListData&& ruleListData)
+{
+    RefPtr compiledContentRuleList = WebCompiledContentRuleList::create(WTFMove(ruleListData));
```

Comment:
How long does this take? It will take up CPU time in the WebProcess potentially in the middle of a page load.

Reply:
It's just copying trivial data and very cheap.

Reply:
This particular line of code seems cheap but not sure about the `addContentExtension` line below.

Reply:
Hmm, sure. I double check this.

Reply:
Checked with Alex and he confirmed this is very cheap and no problem using it in main thread.